framework/controller/controller.gotext

```diff
@@ -69,6 +69,19 @@ func ({{$action.Short}} *{{ $.Pascal }}{{$action.Pascal}}Action) handler(httpRes
 			JSON: response.Status(400).Set("Content-Type", "application/json").JSON(map[string]string{"error": err.Error()}),
 		}
 	}
+	// validate input
+	{{- if (index $action.Params 0).HasValidate }}
+	// single input struct has validate method
+	if err := in.Validate(); err != nil {
```

Comment:
What are your thoughts on calling it `Validate` vs `Valid`? IMO `Validate` sounds more like an action that would return an error, but `Valid` is shorter.

Reply:
I actually like `Valid()` now that you bring it up. I was doing `Validate()` because that was what was in the RFC 15. I don't feel strongly either way.

Reply:
Oh right, let's go with `Valid()` then!